crypto/crypto.go

```diff
@@ -205,6 +205,10 @@ func Sign(hash []byte, prv *ecdsa.PrivateKey) (sig []byte, err error) {
 }
 
 func Encrypt(pub *ecdsa.PublicKey, message []byte) ([]byte, error) {
+	if pub.X == nil || pub.Y == nil {
+		return nil, errors.New("invalid public key")
+	}
```

Comment:
Good question actually. I'll track this down. I assume that somewhere in the code `new(ecdsa.PublicKey)` is done and deserialised however failing because there's no public key resulting in an allocated pub key without `x` and `y`.